.github/workflows/cibuildwheel-before-build.sh

```diff
@@ -20,10 +20,13 @@ curl -L -O "https://github.com/oneapi-src/oneTBB/archive/refs/tags/${TBB_ZIP}"
 unzip -q "${TBB_ZIP}"
 
 #
+echo "PLATFORM=${PLATFORM}"
+echo "ARCHFLAGS=${ARCHFLAGS}"
 EXTRA_CMAKE_ARGS=""
 if [[ "${PLATFORM}" == "macos" ]]; then
+        EXTRA_CMAKE_ARGS="-DCMAKE_INSTALL_PREFIX=/Users/runner/work/tbb-install"
     if [[ ${ARCHFLAGS} == *"arm64"* ]]; then
```

Comment:
The current error is: "         ld: warning: ignoring file '/Users/runner/work/tbb-install/lib/libtbb.12.2.dylib': found architecture 'arm64', required architecture 'x86_64'"

`macos-14` runners are arm64 by default. I think you need to add an `if [[ ${ARCHFLAGS} == *"x86_64"* ]];` similar to this one in order to force tbb compilation for `x86_64`.